tests/shared/drive-constraints.spec.js

```diff
@@ -556,4 +556,229 @@ describe('Shared: DriveConstraints', function() {
+        });
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.ERROR,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.TOO_SMALL
+        });
+      });
+    });
+
+    describe('given the drive is locked', () => {
+
+      it('should return the locked drive error', function() {
+        this.drive.protected = true;
+
+        const result = constraints.getDriveImageCompatibilityStatus(this.drive, {
+          path: '/mnt/disk2/rpi.img',
+          size: 1000000000
+        });
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.ERROR,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.LOCKED
+        });
+      });
+    });
+
+    describe('given the drive is smaller than the recommended size', () => {
+
+      it('should return the smaller than recommended size warning', function() {
+        const result = constraints.getDriveImageCompatibilityStatus(this.drive, {
+          path: '/mnt/disk2/rpi.img',
+          size: 1000000000,
+          recommendedDriveSize: 5000000000
+        });
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.WARNING,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.NOT_RECOMMENDED
+        });
+      });
+    });
+
+    describe('given the image is null', () => {
+
+      it('should return OK', function() {
+        const result = constraints.getDriveImageCompatibilityStatus(this.drive, null);
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.OK,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.OK
+        });
+      });
+    });
+
+    describe('given the drive is null', () => {
+
+      it('should return OK', function() {
+        const result = constraints.getDriveImageCompatibilityStatus(null, {
+          path: '/mnt/disk2/rpi.img',
+          size: 1000000000,
+          recommendedDriveSize: 2000000000
+        });
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.OK,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.OK
+        });
+      });
+    });
+
+    describe('given that everything is wrong', () => {
+
+      it('should return the source drive error by precedence', function() {
+        this.drive.system = true;
+        this.drive.protected = true;
+
+        const result = constraints.getDriveImageCompatibilityStatus(this.drive, {
+          path: `${this.mountpoint}${this.separator}rpi.img`,
+          size: 5000000000,
+          recommendedDriveSize: 5000000000
+        });
+
+        m.chai.expect(result).to.deep.equal({
+          type: constraints.COMPATIBILITY_STATUS_TYPES.ERROR,
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES.CONTAINS_IMAGE
+        });
+      });
+    });
+
+    describe('given everything but source drive is wrong', () => {
```

Comment:
As I mentioned in an earlier comment, I think it's sufficient to just do pair-wise combination checks, to check that for each pair the expected error gets returned first. So something like:
conditionSourceDrive & conditionSystemDrive => errorSourceDrive
conditionSystemDrive & conditionTooSmall => errorSystemDrive
conditionTooSmall & conditionLocked => errorTooSmall
conditionLocked & conditionNotRecommended => errorLocked

A bit like by checking `1 < 2` and `2 < 3` and `3 < 4`, you automatically know that `1 < 3`, `1 < 4`, `2 < 4` without having to explicitly check all combinations ;-)